fbu-app/ViewControllers/HomeScreens/AddHousemateViewController.m

```diff
@@ -0,0 +1,148 @@
+//
+//  AddHousemateViewController.m
+//  fbu-app
+//
+//  Created by sophiakaz on 7/25/19.
+//  Copyright © 2019 lucjia. All rights reserved.
+//
+
+#import "AddHousemateViewController.h"
+#import "PlainRoomateCell.h"
+#import "Parse/Parse.h"
+#import "House.h"
+#import "Persona.h"
+#import "Request.h"
+
+@interface AddHousemateViewController () <UITableViewDataSource, UITableViewDelegate>
+@property (weak, nonatomic) IBOutlet UITableView *tableView;
+- (IBAction)tapAddHousemates:(id)sender;
+@property (nonatomic, strong) NSMutableArray *potentialHousemates;
+@property (nonatomic, strong) NSMutableArray *housematesToAdd;
+@property (weak, nonatomic) IBOutlet UIButton *houseButton;
+
+@end
+
+@implementation AddHousemateViewController
+
+- (void)viewDidLoad {
+    [super viewDidLoad];
+    
+    self.housematesToAdd = [[NSMutableArray alloc] init];
+    
+    self.tableView.dataSource = self;
+    self.tableView.delegate = self;
+    
+    [self setButtonLabel];
+    
+    [self fetchPotentialHousemates];
+    [self.tableView reloadData];
+    // Do any additional setup after loading the view.
+    self.tableView.tableFooterView = [[UIView alloc]
+                                      initWithFrame:CGRectZero];
+}
+
+- (void) setButtonLabel {
+    Persona *persona = [[PFUser currentUser] objectForKey:@"persona"];
+    [persona fetchIfNeeded];
```

Comment:
Instead of using `fetchIfNeeded` use `fetchIfNeededInBackgroundWithBlock:` instead throughout the program in order to avoid "Warning: A long-running operation is being executed on the main thread".